ballerina/context.bal

```diff
@@ -124,12 +124,20 @@ public isolated class Context {
 
     isolated function getNextInterceptor() returns (readonly & Interceptor)? {
         lock {
-            if self.getEngine() is Engine {
-                Engine engine = <Engine>self.getEngine();
-                if engine.getInterceptors().length() > self.nextInterceptor {
-                    readonly & Interceptor next = engine.getInterceptors()[self.nextInterceptor];
-                    self.nextInterceptor += 1;
-                    return next;
+            Engine? engine = self.getEngine();
+            if engine is Engine {
+                (readonly & Interceptor)|(readonly & Interceptor)[] interceptors = engine.getInterceptors();
+                if interceptors is (readonly & Interceptor)[] {
+                    if interceptors.length() > self.nextInterceptor {
+                        readonly & Interceptor next = interceptors[self.nextInterceptor];
+                        self.nextInterceptor += 1;
+                        return next;
+                    }
+                } else {
+                    if self.nextInterceptor == 0 {
+                        self.nextInterceptor += 1;
+                        return interceptors;
+                    }
                 }
```

Comment:
We can remove this (seemingly hard-to-read) code if we return an interceptor array from the annotation processor.

Reply:
Fixed via https://github.com/ballerina-platform/module-ballerina-graphql/pull/1245/commits/0e79ac5618bfd988b2175f4d6fe6e96da485e4af